### PR TITLE
fix(browse): try to prevent overlapping pagination calls

### PR DESCRIPTION
**Pull request by joshuaeilers**
Pagination calls should not run when another call is already running.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/search/sidebar/useBrowsePagination.tsx [hunk 1]
```diff
@@ -30,8 +31,9 @@ const useBrowsePagination = ({ skip }: Props) => {
     const total = latestData?.browseV2?.total ?? -1;
     const done = !!latestData && groups.length >= total;
 
-    const [getBrowseResultsV2, { data, error, refetch }] = useGetBrowseResultsV2LazyQuery({
+    const [getBrowseResultsV2, { data, error, loading, refetch }] = useGetBrowseResultsV2LazyQuery({
         fetchPolicy: 'cache-first',
+        notifyOnNetworkStatusChange: true,
```
Comment (chriscollins3456) [inline datahub-web-react/src/app/search/sidebar/useBrowsePagination.tsx:36]: ah so we need this for `loading` to be up to date right? and that will cause some more re-renders? i'm not too worried about that though.

**Review by chriscollins3456 — APPROVED** [review]
nice

File: datahub-web-react/src/app/search/sidebar/useBrowsePagination.tsx (hunk 1 above)
Comment (joshuaeilers) [reply datahub-web-react/src/app/search/sidebar/useBrowsePagination.tsx:36]: Yeah it'll only swap to loading if we're loading a new page though

Comment (joshuaeilers) [reply datahub-web-react/src/app/search/sidebar/useBrowsePagination.tsx:36]: Probably don't need the loading check with the ref addition, removing.

**Pull request merged**